apps/verticalhome/js/more_apps_navigation.js

```diff
@@ -2,84 +2,103 @@
+    /**
+     * Remove listener of keydown event, when "More Apps" screen is hide
+     */
+    stopListeningKeydownEvents: function () {
       window.removeEventListener('keydown', this);
     },
 
+    /**
+     * Handle keydown event on 'More Apps' screen
+     */
     handleEvent: function (e) {
       var deltaIndex = 0;
-      e.preventDefault();
       switch (e.key) {
         case 'ArrowUp':
           deltaIndex = -2;
-          this.changeSelectElem(deltaIndex);
-          if (!this.isVisible(this.selectElement)) {
-            this.selectElement.scrollIntoView({behavior:"smooth", block: "start"});
-          }
+          this.changeSelectElem(deltaIndex, 'top');
+          e.preventDefault();
           break;
         case 'ArrowDown':
           deltaIndex = 2;
-          this.changeSelectElem(deltaIndex);
-          if (!this.isVisible(this.selectElement)) {
-            this.selectElement.childNodes[0].scrollIntoView({behavior:"smooth", block: "end"});
-          }
+          this.changeSelectElem(deltaIndex, 'bottom');
+          e.preventDefault();
           break;
         case 'ArrowLeft':
           deltaIndex = -1;
-          this.changeSelectElem(deltaIndex);
-          if (!this.isVisible(this.selectElement)) {
-            this.selectElement.scrollIntoView({behavior:"smooth", block: "start"});
-          }
+          this.changeSelectElem(deltaIndex, 'top');
+          e.preventDefault();
           break;
         case 'ArrowRight':
           deltaIndex = 1;
-          this.changeSelectElem(deltaIndex);
-          if (!this.isVisible(this.selectElement)) {
-            this.selectElement.childNodes[0].scrollIntoView({behavior:"smooth", block: "end"});
-          }
+          this.changeSelectElem(deltaIndex, 'bottom');
+          e.preventDefault();
           break;
         case 'Accept':
-          window.removeEventListener('keydown', this);
-          this.selectElement.click();
-          break;
-        case 'BrowserBack':
-        case 'Backspace':
-          app.hideMoreApps();
+          /**
+           * Launch the selected app. Don't need to call hideMoreApps(), because
+           * it is called when the app will close
+           */
+          this.items[this.selectedElemIndex].element.click();
           break;
       }
     },
 
-    changeSelectElem: function (deltaIndex) {
-      this.selectElement.classList.remove('selected');
-      var temp = this.selectedElemIndex + deltaIndex;
-      if (temp > -1 && temp < this.controls.length) {
-        this.selectedElemIndex = temp;
+    /**
+     * Change old selected item to new selected item and scroll it into view if
+     * it doesn't fully visible. If new selectedElemIndex is output of 'items'
+     * array limits, then current selected item don't change
+     */
+    changeSelectElem: function (deltaIndex, behavior) {
+      var trialIndex = this.selectedElemIndex + deltaIndex;
+      if (trialIndex > -1 && trialIndex < this.items.length) {
```

Comment:
minor.
I personally prefer to use "x >= 0 &&  x < size" if statements. Feels a bit more readable for me.